docs/apps/build-system.adoc

```diff
@@ -89,33 +89,51 @@ NOTE:  XP apps deployed via file (locally) are presented with a small blue icon
 
 === Development Mode
 
-Enonic XP supports a so-called Development mode.
-When running in this mode, XP will automatically source JavaScript controllers directly from your project folder.
+[WARNING]
+====
+As the name implies, development mode is intended only for development use and should _not_ be used in production. Doing so would cause both security and performance issues.
+====
 
-This may be convenient if you are developing with pure JavaScript and do not depend on build steps in your project.
+Enonic XP can be run in a special mode known as _development mode_. In this mode, XP will detect changes to your project source files, including JavaScript controllers and registered XML schemas. It does this by reading these files directly from their source locations instead of from the application `.jar` file.
 
-You may activate dev mode when starting XP as follows:
+[NOTE]
+====
+Because of the way that XP works, some changes are not picked up in dev mode: if you add a _new_ content type, XP will not pick it up automatically. This is because content types are registered with XP when _deploying_ an app. However, changes to existing content types _are_ detected. If you'd like to have new content types picked up automatically, try using xref:build-system#_continuous_building[continuous build mode].
+
+This does not affect JavaScript controllers. You can add new controllers and they will be picked up by XP automatically.
```

Comment:
This is where things get complicated for dev mode:
if controllers are written in ES6 or TypeScript they require transpiling. Only vanilla JS controllers can work without `gradle build` step.
Also applies for css vs sass/less

Reply:
You mention it later, so probably it is irrelevant.

Reply:
Why does ES6 require transpiling? From what I understand, JS controllers are somehow interpreted by the Java runtime (using Nashorn?), and this doesn't support the full ES6 feature set. Is this why? 

Not to be pedantic, but I think this is a good distinction to make: when you say "vanilla JS", you mean ES5, is this correct? I think it would be misleading to use "vanilla JS" to refer to pre-ES6 JS in the docs, considering ES6 (and later versions) are still "vanilla".

Reply:
Nashorn supports ES6 partly, there are overview tables that explain what it supports and not. Perhaps we should link to that?

Reply:
That might be useful. Is this documented elsewhere in the XP docs (or other Enonic docs)? I don't think we need to go into intricate detail here, but I think would be helpful to say something along the lines of 

> JavaScript controllers that require some sort of compilation step won't be sourced directly. To find out more about what features are and are not available, check out <these docs>

Reply:
@sigdestad Are these nashorn ES6 compatibility tables available anywhere? I found this [JEP](https://openjdk.java.net/jeps/292), but it doesn't have a clear overview listed.

Reply:
I've used this one: https://kangax.github.io/compat-table/es6/
Nashorn is called JJS afaik.